### PR TITLE
Refactor the voxelize tool

### PR DESCRIPTION
**Pull request by jafyvilla**
Extract the parts that may be common for other tools
and create an abstract class CommandLineApplication that
is used as a base for all the applications.

Remove support of deprecated #target in the volume URI.

### CODE REVIEW

**Comment by jafyvilla** [comment]
Needed for #48

File: apps/voxelize/voxelize.cpp [hunk 1]
```diff
@@ -84,235 +81,135 @@ void _sample( ImageSourcePtr source, const vmml::Vector2ui& frameRange,
...
-{
-    return is >> std::skipws >> vec.x() >> vec.y();
-}
-}
+    bool parse( int argc, char* argv[] ) final
+    {
+        CommandLineApplication::parse( argc, argv );
 
-namespace po = boost::program_options;
+        if( _vm.count( "help" ))
```
Comment (tribal-tec) [inline apps/voxelize/voxelize.cpp]: parse in base class?

**Comment by tribal-tec** [comment]
+1

File: apps/commandLineApplication.h [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+              "          in more volumetric data, and therefore more computation\n"
+              "          time\n"
+              "- showProgress: display progress bar for current voxelization step\n"
+              "                (default: 0/off)\n"
+              "\n"
+              "Parameters for Compartments:\n"
+              "- report: name of the compartment report\n"
+              "          (default: 'voltage'; 'allvoltage' if BlueConfig is BBPTestData)\n"
+              "- dt: timestep between requested frames in milliseconds\n"
+              "      (default: report dt)\n"
```
Comment (hernando) [inline apps/commandLineApplication.h]: You haven't mentioned anywhere in the spec that this parameter doesn't have any effect in the new tool.

**Comment by jafyvilla** [comment]
Commit integrated in #48

**Pull request closed without merging**